From 621eb27d4284fcd71287d2cee94f6156510600fb Mon Sep 17 00:00:00 2001
From: Peter Ujfalusi <peter.ujfalusi@ti.com>
Date: Fri, 12 Feb 2016 11:56:03 +0200
Subject: [PATCH] tty: serial: 8250: omap: Switch to the new
 dmaengine_terminate_sync

Using dmaengine_terminate_all() is deprecated per documentation in
include/linux/dmaengine.h

By using the new dmaengine_terminate_sync() we can avoid dma completion
callback after the channel has been terminated.

Signed-off-by: Peter Ujfalusi <peter.ujfalusi@ti.com>
[nsekhar@ti.com: rebase to v4.4]
Signed-off-by: Sekhar Nori <nsekhar@ti.com>
---
 drivers/tty/serial/8250/8250_omap.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/tty/serial/8250/8250_omap.c b/drivers/tty/serial/8250/8250_omap.c
index a2c0734..eb74c40 100644
--- a/drivers/tty/serial/8250/8250_omap.c
+++ b/drivers/tty/serial/8250/8250_omap.c
@@ -737,7 +737,7 @@ static void __dma_rx_do_complete(struct uart_8250_port *p, bool error)
 
 	dma->rx_running = 0;
 	dmaengine_tx_status(dma->rxchan, dma->rx_cookie, &state);
-	dmaengine_terminate_all(dma->rxchan);
+	dmaengine_terminate_sync(dma->rxchan);
 
 	count = dma->rx_size - state.residue;
 
-- 
2.8.1

